zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -2271,7 +2271,14 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
   });
 
   $scope.$on('appendParagraphOutput', function(event, data) {
-    if ($scope.paragraph.id === data.paragraphId) {
+    /* It has been observed that append events
+     * can be errorneously called even if paragraph
+     * execution has ended, and in that case, no append
+     * should be made. Also, it has been observed that this
+     * could be called in PENDING state as well
+     */
+    if ($scope.paragraph.id === data.paragraphId &&
+       ($scope.paragraph.status === 'RUNNING' || $scope.paragraph.status === 'PENDING')) {
```

Comment:
What I observed was that in between the pending and the running state, some (I observed 1) append-output events were called. If we miss those, then while the paragraph is running, we would see that some intial-result line/s is/are missing.
Also, when the para execution is finished, the complete note is sent again (with results), so we would see the correct result.